kvbc/src/ReplicaImp.cpp

```diff
@@ -83,13 +83,15 @@ void ReplicaImp::createReplicaAndSyncState() {
       &m_replicaConfig, m_cmdHandler, m_stateTransfer, m_ptrComm, m_metadataStorage);
   if (!isNewStorage && !m_stateTransfer->isCollectingState()) {
     uint64_t removedBlocksNum = replicaStateSync_->execute(
-        logger, *m_bcDbAdapter, m_appState->m_lastReachableBlock, m_replicaPtr->getLastExecutedSequenceNum());
+        logger, *m_bcDbAdapter, m_appState->getLastReachableBlockNum(), m_replicaPtr->getLastExecutedSequenceNum());
     m_lastBlock -= removedBlocksNum;
-    m_appState->m_lastReachableBlock -= removedBlocksNum;
+    auto t = std::dynamic_pointer_cast<BlockchainAppState>(m_appState);
```

Comment:
what if the m_appState is a `RoRAppState` ?

Reply:
This code is executed only on tjhe non-ro replica, see line 62 - only for non-ro replica this function will be called

Reply:
As mentioned in a prior comment: This downcast would not be needed if there was a separate ReplicaImp for the read only replica.